tools/trace_synthesizer/synthesizer.py

```diff
@@ -94,7 +94,7 @@ def generate(args):
             pattern = p * repetitions
 
             invArr.extend(pattern)
-        else:
+        elif mode == 2:
```

Comment:
Is that indended to be changed in this PR? And also, I don't see much reason to change that at all.